models/storageProvider.go

```diff
@@ -3,7 +3,7 @@ package models
 //StorageProvider is how a chunk storage provider should be implemented
 type StorageProvider interface {
 	Setup() error
-	Save(filePath string, retryCount int) (string, error)
+	Save(filePath string, retryCount int) (*string, error)
```

Comment:
Okay, now I'm curious why this change. Usually pointers are only used to enable mutability on an object or when marshelling and you don't want that value marshelled

Reply:
No problem, I just found it easier to read my own code checking for nil.  Probably makes sense to rely more on error anyway, I'll change it back.